### PR TITLE
Shanbady/topic channel page header fixes

### PR DESCRIPTION
**Pull request by shanbady**
### What are the relevant tickets?

Closes https://github.com/mitodl/hq/issues/4542 and https://github.com/mitodl/hq/issues/4540 and https://github.com/mitodl/mit-open/issues/1045

### Description (What does it do?)
This PR fixes 3 sets of issues. Some style changes to the channel page banner make it more inline with the figma (see https://github.com/mitodl/hq/issues/4542 and https://github.com/mitodl/hq/issues/4540) and the addition of a configurable dark overlay to channel page banners (https://github.com/mitodl/mit-open/issues/1045)

### Screenshots (if appropriate):

- [ ] Desktop screenshots
Screenshot 2024-06-12 at 9 43 40 AM
Screenshot 2024-06-12 at 9 43 55 AM
Screenshot 2024-06-12 at 9 48 23 AM

- [ ] Mobile width screenshots
Screenshot 2024-06-12 at 9 46 13 AM
Screenshot 2024-06-12 at 9 50 47 AM
Screenshot 2024-06-12 at 9 51 27 AM

### How can this be tested?
1. checkout this branch and make sure units/offerors and topics are loaded 
  ```
   python manage.py update_offered_by
   python manage.py backpopulate_resource_channels --overwrite --unit
   python manage.py update_platforms
   python manage.py update_topics
   ```
3. login and visit the various channel pages like http://od.odl.local:8063/c/unit/see http://od.odl.local:8063/c/unit/xpro http://od.odl.local:8063/c/unit/bootcamps http://od.odl.local:8063/c/department/history and http://od.odl.local:8063/c/topic/physics
4. validate that the pages look like the figma mocks: 

desktop:
  - https://www.figma.com/proto/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?page-id=107%3A2479&node-id=3774-68962&viewport=21%2C-3032%2C0.89&t=Hy7uaWCxhWB7TXwN-1&scaling=min-zoom&starting-point-node-id=2197%3A31345
  - https://www.figma.com/proto/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?page-id=107%3A2479&node-id=2197-31446&viewport=21%2C-3032%2C0.89&t=Hy7uaWCxhWB7TXwN-1&scaling=min-zoom&starting-point-node-id=2197%3A31446
   

mobile:
  - https://www.figma.com/proto/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?page-id=107%3A2479&node-id=4083-96092&viewport=21%2C-3032%2C0.89&t=Hy7uaWCxhWB7TXwN-1&scaling=min-zoom&starting-point-node-id=2197%3A31345
  - https://www.figma.com/proto/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?page-id=107%3A2479&node-id=2197-31345&viewport=21%2C-3032%2C0.89&t=Hy7uaWCxhWB7TXwN-1&scaling=min-zoom&starting-point-node-id=2197%3A31345&show-proto-sidebar=1

### CODE REVIEW

**Review by jkachel — CHANGES_REQUESTED** [review]
Couple things I noticed:

Desktop:
- Top and bottom padding on the banner container is off - looks like 24px where the Figma design has 48px top and 64px bottom

Mobile:
- Top and bottom padding on the "left" side is also off - should be 32px here
- Interior padding on the info box is off - should be 16px where it's 32px
- The info box too is too big and goes off the screen.

Here's a screenshot in responsive view (Firefox 126, macOS, with the selector set to Galaxy Note 20):
image

Otherwise looks pretty good.

**Comment by shanbady** [comment]
@jkachel pushed up a fix for the header padding and panel sizing

**Comment by Ferdi** [comment]
A couple of comments, feel free to decide whether they should be considered here or on a different PR.   cc @steven-hatch  @Bilal (I cant @ Bilal for some reason @pdpinch )


Breadcrumbs 

`home/units/ocw`

Home should be a link the the home page
Units should be a link to the unit compare page
ocw should be the long name (MIT OpenCourseWare ?)

I think the ordering of the metadata  (in the info box) is important. For example "Offerings" makes sense to be first and "more information" to be last. @shanbady is it hard to preserve the ordering ?

In the design  some of the metadata:value is presented in one line  (E.g: `Fee: Free`) and some in separate lines e.g.

```
Offerings:
Learning Materials
```

but in the implementation, it's always the later.    I dont have a preference (@steven-hatch ?) but the only thing I would say is that if you go with all in separate lines , then no need for `:`

**Comment by shanbady** [comment]
@Ferdi we have a way of preserving the ordering based off of the field which I can add to this pr. 

I can also linkify the breadcrumbs but had some questions 
  - it wasnt clear from figma if there is a hover state on them
  - at the moment the breadcrumb is not a shared component. for example the /units page has its own breadcrumb and the text is not consistent https://mitopen-rc.odl.mit.edu/units/ - if we want to fully standardize this into a shared component it might be worth a separate pr

**Comment by shanbady** [comment]
@Ferdi regarding the OCW in the nav - this appears to be a data issue. the nav text comes from what is supposed to be the "long title" but for the OCW offeror object [it is "ocw"](https://api.mitopen-rc.odl.mit.edu/admin/learning_resources/learningresourceofferor/ocw/change/).

**Review by jkachel — CHANGES_REQUESTED** [review]
Looks better on desktop and the info box is better on mobile. Just a few more spacing things:

- On desktop, the info box has 16px padding where it should be 32px
- On mobile:
   - the space under the infobox is 64px where it should be 16px
   - the space above the infobox should be 48px (16+32 in the figma) where now it is 24px
   - the space above the breadcrumb is 48px where it should be 32px
   - the spacing between the elements in the "left"/content pane should be 16px, except above the logo where it's 32px

**Comment by Ferdi** [comment]
@shanbady 

> at the moment the breadcrumb is not a shared component. for example the /units page has its own inline breadcrumb https://mitopen-rc.odl.mit.edu/units/ - if we want to fully standardize this into a shared component it might be worth a separate pr

I'd consider doing a shared component later.

**Comment by steven-hatch** [comment]
Regarding Breadcrumbs - Yes, these should eventually be in component form. https://www.smashingmagazine.com/2022/04/breadcrumbs-ux-design/

`Home / Units / OpenCourseWare`
- Home should be a link the the home page ✓ 
- Units should be a link to the unit compare page ✓ 
- OCW - This can OCW or OpenCourseWare. The breadcrumb does not need to match the H1 but it should be similar - This should also not be a link, but just solid text or styled in a way that has reduced visual importance. I will add breadcrumb component into design system figma so we have a consistent source of truth because Bilal and I have been added this in without component structure. 

Regarding DataPairs for Units Page
Lets do ...

**Offerings**
Learning Materials

Because they are on separate lines. Thanks!

**Comment by shanbady** [comment]
pinned the ordering and remove the ":" from the side panel
Screenshot 2024-06-13 at 12 46 26 PM

**Comment by pdpinch** [comment]
> @Ferdi regarding the OCW in the nav - this appears to be a data issue. the nav text comes from what is supposed to be the "long title" but for the OCW offeror object [it is "ocw"](https://api.mitopen-rc.odl.mit.edu/admin/learning_resources/learningresourceofferor/ocw/change/).

@shanbady the unit/offeror data should match what's in this document: https://docs.google.com/document/d/1-Kyull4biR4XhSsPK6ZLHhqEOLyUEdnQjP6Xlvqe2zI/edit#heading=h.1z6344o0szco

According to the doc, the title should be "OpenCourseWare" 

I opened a PR https://github.com/mitodl/mit-open/pull/1079 that I think will correct it, but I'm not sure. If it needs work, please take over the branch from me and fix it.

File: frontends/mit-open/src/pages/FieldPage/FieldPageSkeleton.tsx [hunk 1]
```diff
@@ -68,10 +67,11 @@ const FieldSkeletonProps: React.FC<FieldSkeletonProps> = ({
         displayConfiguration?.banner_background ??
         "/static/images/background_steps.jpeg"
       }
-      alt=""
       omitBackground={field.isLoading}
+      backgroundSize="2000px auto"
+      dim={30}
       bannerContent={
-        <Container sx={{ my: 2, py: "48px" }}>
+        <Container sx={{ pt: "48px", pb: "64px" }}>
```
Comment (jkachel) [inline frontends/mit-open/src/pages/FieldPage/FieldPageSkeleton.tsx:74]: OK for desktop but on mobile this should be 32px/16px

File: frontends/mit-open/src/pages/FieldPage/FieldPageSkeleton.tsx [hunk 2]
```diff
@@ -42,7 +41,7 @@ const NAV_PATH: { [key: string]: string } = {
 
 const NavText = styled(Typography)(({ theme }) => ({
   color: theme.custom.colors.lightGray2,
-  marginBottom: "10px",
+  marginBottom: "16px",
```
Comment (jkachel) [inline frontends/mit-open/src/pages/FieldPage/FieldPageSkeleton.tsx:44]: Should be 32px on mobile, I think

File: frontends/mit-open/src/pages/FieldPage/FieldPageSkeleton.tsx [hunk 3]
```diff
@@ -94,13 +94,18 @@ const FieldSkeletonProps: React.FC<FieldSkeletonProps> = ({
...
+                    order: 1,
+                    width: "50%",
+                  }}
                 >
                   <Box
                     display="flex"
                     flexDirection="row"
                     alignItems="center"
-                    sx={{ flexGrow: 1, flexShrink: 0, order: 1, mt: 3 }}
+                    sx={{ flexGrow: 1, flexShrink: 0, order: 1, py: "24px" }}
```
Comment (jkachel) [inline frontends/mit-open/src/pages/FieldPage/FieldPageSkeleton.tsx:108]: For mobile I think this works best as 8px bottom, 0 top (with the 32px bottom padding on the `NavText` noted above.

File: frontends/mit-open/src/pages/FieldPage/FieldPageSkeleton.tsx [hunk 4]
```diff
@@ -209,11 +214,11 @@ const FieldSkeletonProps: React.FC<FieldSkeletonProps> = ({
                     flexDirection="row"
                     alignItems="end"
                     sx={{
```
Comment (jkachel) [inline frontends/mit-open/src/pages/FieldPage/FieldPageSkeleton.tsx:216]: Should be some more space here - `margin-top: 8px` and `margin-bottom: 48px` seemed to look ok to me

**Review by jkachel — CHANGES_REQUESTED** [review]
The spacing on mobile is still just a bit off on mobile. I added some comments - they're based on some futzing with the layout in the CSS inspector so may need a bit of adjustment in situ.

File: frontends/mit-open/src/pages/FieldPage/FieldPageSkeleton.tsx (hunk 2 above)
Comment (shanbady) [reply frontends/mit-open/src/pages/FieldPage/FieldPageSkeleton.tsx:44]: fixed

File: frontends/mit-open/src/pages/FieldPage/FieldPageSkeleton.tsx (hunk 3 above)
Comment (shanbady) [reply frontends/mit-open/src/pages/FieldPage/FieldPageSkeleton.tsx:108]: fixed

File: frontends/mit-open/src/pages/FieldPage/FieldPageSkeleton.tsx (hunk 4 above)
Comment (shanbady) [reply frontends/mit-open/src/pages/FieldPage/FieldPageSkeleton.tsx:216]: is this in the mobile view? also i was wondering if you had text populated in the banner. when i add this locally it seems to look like too much of a gap

Comment (jkachel) [reply frontends/mit-open/src/pages/FieldPage/FieldPageSkeleton.tsx:229]: Yup - this is with data in there as well (I copied the config for OCW out of RC). Here's a screenshot:

image

(The white lines here are from xScope.)

**Review by jkachel — APPROVED** [review]
LGTM 👍

**Pull request merged**